src/config.cpp

```diff
@@ -661,7 +661,13 @@ bool N64Recomp::Context::import_reference_context(const N64Recomp::Context& refe
 
     // Copy the functions from the reference context into the reference context's function map.
     for (const N64Recomp::Function& func_in: reference_context.functions) {
-        if (!add_reference_symbol(func_in.name, func_in.section_index, func_in.vram, true)) {
+        // Rename if necessary
+        std::string name = func_in.name;
+        if (N64Recomp::renamed_funcs.contains(name)) {
```

Comment:
The way this works looks good, it just needs to also check ignored_funcs and reimplemented_funcs as well like here: https://github.com/N64Recomp/N64Recomp/blob/53ffee96fd6666f10a28de5712fd24a29253a4da/src/main.cpp#L415